SWIPs/swip-streaming-of-feeds.md

```diff
@@ -0,0 +1,148 @@
+
+The current worked out solutions of handling a content stream (Book of Swarm, Subscriptions, page 112) approach the update tracking by:
+- push notifications about the update (PSS)
+- epoch-based indexing for sporadically updated feeds
+- polling for almost ideal periodic updates (frequent updates)
+
+Though the book mentiones also _periodic feeds_ for calculating index of feed starting from a specific well-known time, 
+but it does not touch the scenario what if the updates are _parlty sporadic_ and those are may not available or not uploaded.
+
+This requires an **indexing method** that extends the `periodic feeds` indexing and a **specific retrieval method** to handle the holes in the update stream.
+
+The best approach to get the closest feed of an arbitrary time is using epoch-based feeds, but its downside - additionaly to that I summarized [above](##-Motivation) - is after every writing the retrieval is slower. 
+It is similar to that analogy when a hard drive is fragmenting because of the many frequent writes within one sector (base fragment of the epoch time lookup). 
+This proposed feed indexing method is the opposite as at lookups: 
+if there is expected amount of writes periodically, the retrieval of the data is faster, that even can be _O(1)_. 
+Compared to the epoch-based feeds, basically the length of the base segment is arbitrary, and it encourages the user to make periodic uploads and stick to this base segment instead of sporadic uploading in exchange for better retrieval time.
+
+The downside is within the concerted update time period the content creator cannot update the state of the feed - although it is possible to overwrite the content of a feed by uploading it again with different payload, but on download-side it is possible to still to get back the old one. 
+This problem can be mitigated by changing the registry of the feed, from where the consumers get the initial metadatas of the feed: 
+- the initial timestamp (`T0`) should be changed to that point from where the time period changes (`T1`)
+- change the new time period (`Δ2`) from the old one (`Δ1`) for state uploads
+
+If the registry settled on blockchain in a smart contract, then it can have a defined event on metadata change of the content, on which the clients can listen.
+Thereby, the consumers of the feed can immediately react to the upload frequence change and they can poll according to the new rules.
+If somehow the blockchain listening on client side is not suitable, it is possible to put the `uploading time period` and `initial timestamp` metadatas into all of the feed stream updates so that the consumers can sync to the stream after `MAX(Δ1,Δ2)` time if `MAX(Δ1,Δ2) % MIN(Δ1,Δ2) = 0` and  there is one `k` and `m` positive integer where `T0 + (k * Δ1) = T0 + (m * Δ2) = T1`
+
+Though it is stated this approach does not address downloading the whole feed stream, it is still possible:
```

Comment:
ets leave this aside. i think historical data should be time-indexed in a POT data structure, put the current root hash in the feed update